week-2/stretch/till.js

```diff
@@ -30,3 +30,15 @@ const totalAmount = totalTill(till);
 // c) What does coin * quantity evaluate to inside the for...of loop?
 
 // d) Write a test for this function to check it works and then fix the implementation of totalTill
+function totalTill(till) {
+  let total = 0;
+
+  for (const [coinName, quantity] of Object.entries(till)) {
+    const coinValue = Number(coinName.replace("p", "")); // Parse coin value from string
+    total += coinValue * quantity;
+  }
+
+  return `£${total / 100}`;
```

Comment:
Very good, and the explaination is excellent. Good on you for giving the strech work a crack too and doing so well at it  :)